### PR TITLE
feat(ingest): power-bi - make ownership ingestion optional

### PR DESCRIPTION
**Pull request by mohdsiddique**

### CODE REVIEW

**Comment by shirshanka** [comment]
@mohdsiddique : seems like there is a conflict with the main branch, so you might need to refresh your PR.

**Comment by mohdsiddique** [comment]
@shirshanka merge conflict is resolved now.

**Comment by shirshanka** [comment]
@mayurinehate : can you review this?

**Comment by jjoyce0510** [comment]
What's the status on this one?

File: metadata-ingestion/src/datahub/ingestion/source/powerbi.py [hunk 1]
```diff
@@ -1311,23 +1356,13 @@ def report_charts_dropped(self, view: str) -> None:
 @platform_name("PowerBI")
 @config_class(PowerBiDashboardSourceConfig)
 @support_status(SupportStatus.CERTIFIED)
-@capability(SourceCapability.OWNERSHIP, "Enabled by default")
+@capability(SourceCapability.OWNERSHIP, "Optionally enabled via configuration")
```
Comment (hsheth2) [inline metadata-ingestion/src/datahub/ingestion/source/powerbi.py:1359]:
```suggestion
@capability(SourceCapability.OWNERSHIP, "On by default but can disabled by configuration")
```

File: metadata-ingestion/src/datahub/ingestion/source/powerbi.py [hunk 2]
```diff
@@ -104,41 +104,45 @@ class Constant:
...
-
-    scope: str = "https://analysis.windows.net/powerbi/api/.default"
-    base_url: str = "https://api.powerbi.com/v1.0/myorg/groups"
-    admin_base_url = "https://api.powerbi.com/v1.0/myorg/admin"
-    authority = "https://login.microsoftonline.com/"
-
-    def get_authority_url(self):
-        return f"{self.authority}{self.tenant_id}"
+    # Enable/Disable extracting ownership information of Dashboard
+    override: CapabilitiesOverride = CapabilitiesOverride(extractOwnership=True)
```
Comment (hsheth2) [inline metadata-ingestion/src/datahub/ingestion/source/powerbi.py:137]: I'm in favor of not introducing another level of nesting into the config. We have this pattern of bool flags throughout the codebase, for example the `include_views` option on the SQL sources. In those cases, it'd be pretty confusing to have `include_views` nested under a capabilities object but have the `view_patterns` still be top-level. Overall, I think we should keep the config flat unless there's a compelling reason not to.

**Review by hsheth2 — CHANGES_REQUESTED** [review]

File: metadata-ingestion/docs/sources/powerbi/powerbi_recipe.yml [hunk 3]
```diff
@@ -5,17 +5,23 @@ source:
...
-    # Azure AD App client identifier
+    # Azure AD Application identifier
     client_id: foo
     # Azure AD App client secret
     client_secret: bar
+    # override the capabilities settings if needed
+    override:
+      # Enable / Disable ingestion of user information of dashboards
+      # Default is true
+      extractOwnership: true
```
Comment (hsheth2) [inline metadata-ingestion/docs/sources/powerbi/powerbi_recipe.yml:18]: update this

File: metadata-ingestion/docs/sources/powerbi/powerbi_recipe.yml [hunk 4]
```diff
@@ -5,17 +5,23 @@ source:
...
     # dataset_type_mapping is fixed mapping of Power BI datasources type to equivalent Datahub "data platform" dataset
     dataset_type_mapping:
         PostgreSql: postgres
         Oracle: oracle
 
 
+
```
Comment (hsheth2) [inline metadata-ingestion/docs/sources/powerbi/powerbi_recipe.yml:25]: nit - remove

File: metadata-ingestion/src/datahub/ingestion/source/powerbi.py [hunk 5]
```diff
@@ -104,41 +101,39 @@ class Constant:
...
+    client_secret: str = pydantic.Field(description="Azure app client secret")
     # timeout for meta-data scanning
-    scan_timeout: int = Field(
-        default=60,
-        description="time in seconds to wait for Power BI metadata scan result.",
+    scan_timeout: int = pydantic.Field(
+        default=60, description="timeout for PowerBI metadata scanning"
+    )
+    # Enable/Disable extracting ownership information of Dashboard
+    include_ownerships: bool = pydantic.Field(
```
Comment (hsheth2) [inline metadata-ingestion/src/datahub/ingestion/source/powerbi.py:126]:
```suggestion
    extract_ownership: bool = pydantic.Field(
```

**Review by hsheth2 — CHANGES_REQUESTED** [review]
minor nits but should be good to merge after this

**Review by hsheth2 — APPROVED** [review]

**Pull request merged**